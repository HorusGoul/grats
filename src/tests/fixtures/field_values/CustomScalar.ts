/** @GQLScalar */
type MyString = string;

/** @GQLType */
export default class Query {
  /** @GQLField */
  hello(): MyString {
    return "Hello world!";
  }
}
