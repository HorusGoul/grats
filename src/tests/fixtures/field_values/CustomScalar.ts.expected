-----------------
INPUT
----------------- 
/** @gqlScalar */
type MyString = string;

/** @gqlType */
export default class SomeType {
  /** @gqlField */
  hello(): MyString {
    return "Hello world!";
  }
}

-----------------
OUTPUT
-----------------
-- SDL --
directive @asyncIterable on FIELD_DEFINITION

directive @propertyName(name: String!) on FIELD_DEFINITION

directive @exported(tsModulePath: String!, functionName: String!, argCount: Int!) on FIELD_DEFINITION

scalar MyString

type SomeType {
  hello: MyString
}
-- TypeScript --
import { GraphQLSchema, GraphQLScalarType, GraphQLObjectType } from "graphql";
function getSchema(): GraphQLSchema {
    const MyStringType: GraphQLScalarType = new GraphQLScalarType({
        name: "MyString"
    });
    const SomeTypeType: GraphQLObjectType = new GraphQLObjectType({
        name: "SomeType",
        fields() {
            return {
                hello: {
                    name: "hello",
                    type: MyStringType
                }
            };
        }
    });
    return new GraphQLSchema({
        types: [MyStringType, SomeTypeType]
    });
}
export { getSchema };
